tests/testthat/test-error.R

```diff
@@ -6,21 +6,30 @@ test_that("construct error", {
     pkgapi_error_data(list(a = NULL)),
     list(list(error = jsonlite::unbox("a"), detail = NULL)))
   expect_equal(
-    pkgapi_error_data(list(a = "b")),
+    pkgapi_error_data(list(a = list(detail = "b"))),
     list(list(error = jsonlite::unbox("a"), detail = jsonlite::unbox("b"))))
   expect_equal(
-    pkgapi_error_data(list(a = "b", c = NULL)),
+    pkgapi_error_data(list(a = list(detail = "b"), c = NULL)),
     list(list(error = jsonlite::unbox("a"), detail = jsonlite::unbox("b")),
          list(error = jsonlite::unbox("c"), detail = NULL)))
+  trace <- c(jsonlite::unbox("this is"), jsonlite::unbox("the trace"))
+  expect_equal(
+    pkgapi_error_data(list(a = list(
+      detail = "test",
+      key = jsonlite::unbox("fake_key"),
+      trace = trace
```

Comment:
Here you're using a vector but the current implementation of `pkgapi_stop` would prevent this I think. Regardless of any change to that function, does this mean we lack an test here?

Reply:
We're only using `assert_character` on the additional data - not `assert_scalar_character`

Reply:
Oops - I had changed that previously and not updated my comment which says it has to be a "scalar character"

Reply:
that would be it!